### PR TITLE
Ignore some files in npm pack

### PR DESCRIPTION
**Pull request by fregante**
The `.github` folder appears in the published/installed package, along other unrelated files.

[`files`](https://docs.npmjs.com/cli/v6/configuring-npm/package-json) excludes unnecessary files (package.json, license, readme are always included)

### CODE REVIEW

**Review by ljharb — CHANGES_REQUESTED** [review]
I can certainly exclude `.github`. However:
1. the "files" property is dangerous, and `.npmignore` is preferred.
1. tests must be included, as `npm explore foo && npm install && npm test` should always work

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.github
+.eslintrc
+.eslintignore
```
Comment (ljharb) [inline .npmignore:3]: omitting these will cause `npm test` to fail

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+CHANGELOG.md
```
Comment (ljharb) [inline .npmignore:4]: the changelog absolutely belongs in the published package. offline access matters.

**Review by ljharb — CHANGES_REQUESTED** [review]
Typically this is done by `cp .gitignore .npmignore`, and then you add `.github` to the bottom.

File: .npmignore (hunk 1 above)
Comment (fregante) [reply .npmignore:3]: It doesn't look like listing is included in the tests

**Comment by fregante** [comment]
> Typically this is done by `cp .gitignore .npmignore`, and then you add `.github` to the bottom.

Looks like a lot of noise that can be avoided by just using `files`, but that's just my opinion

**Comment by ljharb** [comment]
Indeed; it's unfortunate there's no exclusion list format that can be a diff on top of `.gitignore`. A safelist format is dangerous.

**Review by ljharb — APPROVED** [review]
rebased, and removed eslint configs.

File: .npmignore (hunk 1 above)
Comment (ljharb) [reply .npmignore]: `npm test` runs `npm run pretest` which runs `eslint`.

**Pull request merged**